www/js/coolhole-util.js

```diff
@@ -301,3 +320,166 @@ function playSound( sfxLibItem ) {
 //-----------------------------------------------------------
 // [END] SOUND EFFECTS
 //-----------------------------------------------------------
+
+
+
+
+//-----------------------------------------------------------
+// [START] EMOTE EFFECTS
+//-----------------------------------------------------------
+/**
+ * Applies effects for "returnFire" emote. If the returnFire emote is in the chat message, it shoots the previous chat message.
+ *   If the previous chat message is normal, the previous chat message gets "destroyed".
+ *   If the previous chat message is gold and the current is normal, the current chat message is destroyed, and play a "tink" sound effect.
+ *   If the previous chat message is gold and the current is gold, neither message is destroyed, and play a "tink" sound effect.
+ * @param {Object} div jquery div object of the entire chat message (timestamp, name, message)
+ * @param {Object} jqueryChatSpan jquery span object of only the chat message
+ */
+function checkReturnFire(div, jqueryChatSpan) {
+    // This prevents shadowmuted people from returnFire on non-shadowmuted people (only really applies to people who can actually SEE shadowmutes, like mods/admins viewing the chat)
+    if (div.hasClass("chat-shadow")) 
+        return;
+
+    if (jqueryChatSpan.find("img[title='returnfire']").length > 0) {
+        // Only allow shooting to chat messasges
+        let prevChat = div.prev("div[class*=chat-msg]");
+
+        if(prevChat.length > 0) {
+            let shooterIsGold = false;
+            let victimIsGold = false;
+            let playSound = false;
+    
+            shooterIsGold = div.find("span.text-lottery").length > 0;
+            victimIsGold = prevChat.find("span.text-lottery").length > 0;
+
+            if(shooterIsGold && victimIsGold) {
+                playSound = true;
+            }
+            else if (!shooterIsGold && victimIsGold) {
+                playSound = true;
+                div.addClass("shotKilled");
+            }
+            else {
+                prevChat.addClass("shotKilled");
+            }
```

Comment:
nit: I think logic could be cleaned up a bit although it doesn't read exactly like the jsdoc comment does with this
```suggestion
            const shooterIsGold = div.find("span.text-lottery").length > 0;
            const victimIsGold =  prevChat.find("span.text-lottery").length > 0;
            const playSound = victimIsGold;
            
            if (!victimIsGold) {
                div.addClass("shotKilled");
            }
            else if (!shooterIsGold) {
                prevChat.addClass("shotKilled");
            }
```

Reply:
The way it was originally written feels more clear to me.
It just goes through the 4 combinations of "isShooterGold" and "isVictimGold" (with 2 combinations being combined in the else). Originally I had
It may not be as efficient or succinct, but its more clear on what is supposed to happen based on the 2 inputs. Hope you understand.

Reply:
Agree to disagree. I think this clearly describes:
1. what `shooterIsGold`, `victimIsGold`, and `playSound`'s states are at the start of the function and aren't mutated later on
2. describes the two scenarios for when we add the `shotKilled` class and shows that the logic for playing a sound is really just dependent on the victim being gold

But it's mostly me being pedantic for the sake of the review. Doesn't bother me if it stays this way